xresolver/xresolver_test.go

```diff
@@ -0,0 +1,101 @@
+package xresolver
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestClient(t *testing.T) {
+	assert := assert.New(t)
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintln(w, "Hello, client")
+	}))
+	defer ts.Close()
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			DialContext: NewResolver(nil).DialContext,
+		},
+	}
+
+	req, err := http.NewRequest("GET", ts.URL, nil)
+	assert.NoError(err)
+
+	res, err := client.Do(req)
+	assert.NoError(err)
+	assert.Equal(200, res.StatusCode)
+}
+
+type testCustomLookUp struct {
```

Comment:
Consider replacing this with a mock.  If `Lookup` is an interface, you can easily use `stretchr/testify` to create a mock that plays well with the `golang` `testing` package.